pkg/config/config.go

```diff
@@ -568,6 +568,15 @@ type ImageConfig struct {
 	// containing credentials necessary for pulling images from secure
 	// registries.
 	GlobalAuthFile string `toml:"global_auth_file"`
+	// NamespacedAuthDir is the root path for pod namespace-separated
+	// auth files. The final auth file will be <NAMESPACED_AUTH_DIR>/<NAMESPACE>-<IMAGE_NAME_SHA256.json.
```

Comment:
`<NAMESPACED_AUTH_DIR>/<NAMESPACE>/<IMAGE_NAME_SHA256>.json` may be better in case users want to have directory based access control, or am I paranoid?